sky/skylet/providers/kubernetes/node_provider.py

```diff
@@ -237,10 +239,42 @@ def create_node(self, node_config, tags, count):
                         # If container_statuses is None, then the pod hasn't
                         # been scheduled yet.
                         all_ready = False
-            if all_ready:
+
+            # check if all the pods and containers within the pods are running
+            if  all([ pod.status.phase == "Running" for pod in pods]) \
+                and all([container.state.running for pod in pods for container in pod.status.container_statuses]):
+                pods_and_containers_running = True
+
+            if all_ready and pods_and_containers_running:
                 break
             time.sleep(1)
 
+        # Kubernetes automatically populates containers with critical
+        # environment variables, such as those for discovering services running
+        # in the cluster and CUDA/nvidia environment variables. We need to
+        # update task environment variables with these env vars. This is needed
+        # for GPU support and service discovery.
+        # See https://github.com/skypilot-org/skypilot/issues/2287 for
+        # more details.
+        # Capturing env. var. from the pod's runtime and writes them to
+        # /etc/profile.d/ making them available for all users in future
+        # shell sessions.
+        set_k8s_env_var_cmd = [
+            '/bin/sh', '-c',
+            'printenv | awk \'{print "export " $0}\' > ~/k8s_env_var.sh && sudo mv ~/k8s_env_var.sh /etc/profile.d/k8s_env_var.sh'
```

Comment:
`sudo` may not be present on all images. Can we try mv without sudo first, and use sudo if it fails?